# Assesing microbiome profiles of celiac disease patients
Case-control study of the microbiome in celiac disease through the analysis of 16S high-throughput sequencing data available in public databases. Data of nine studies acounting for 465 samples taken from 4 different parts of the body (duodenum, saliva, stool and pharynx) was analized to asses the differences in alpha and betha diversity, differential abundance at genus level and correlation coeficients. 
The metabolic potential of the microbiome was infered and differential abundance of genes and pathways as well as correlation coeficients
wereobtained. Special enphasisis was stated on enzymes involved in Short-Chain Fatty Acids production and proteases with endoprolyl peptidase activity. 

here you can find the code used to perform the analysis.

1. Biological variants detection with Dada2
2. Taxonomic Assignment with Dada2 and SILVA database
3. Diversity Analysis with Phyloseq and Vegan
4. Differential Abundance with DESeq2 included in phyloseq
5. Correlation Analysis were done in MycrobiomeAnalyst online tool https://www.microbiomeanalyst.ca/
